spark/spark-3.5/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/GeoParquetWriteSupport.scala

```diff
@@ -172,9 +186,10 @@ class GeoParquetWriteSupport extends WriteSupport[InternalRow] with Logging {
         val bbox = if (geometryTypes.nonEmpty) {
           Seq(columnInfo.bbox.minX, columnInfo.bbox.minY, columnInfo.bbox.maxX, columnInfo.bbox.maxY)
         } else Seq(0.0, 0.0, 0.0, 0.0)
-        columnName -> GeometryFieldMetaData("WKB", geometryTypes, bbox)
+        val crs = geoParquetColumnCrsMap.get(columnName).orElse(defaultGeoParquetCrs)
```

Comment:
According to the geoparquet spec, if a CRS is not given, then the crs field of the geoparquet should be omitted assuming this is CRS84. If a CRS is set null, then the crs field of the geoparquet should be set null.

The question is how a user can specify the null here. I think the user need to manually specify `geoparquet.crs` -> `null` via the options. What do you think?

Reply:
Currently the `crs` field is always present (written as `null`). This is because early version of geopandas (for example, 0.10.2 used by python tests) cannot read geoparquet files without `crs` metadata:

```python
>>> geopandas.read_parquet('gp_sample2.parquet')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/opt/homebrew/lib/python3.11/site-packages/geopandas/io/arrow.py", line 461, in _read_parquet
    return _arrow_to_geopandas(table)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/homebrew/lib/python3.11/site-packages/geopandas/io/arrow.py", line 318, in _arrow_to_geopandas
    _validate_metadata(metadata)
  File "/opt/homebrew/lib/python3.11/site-packages/geopandas/io/arrow.py", line 162, in _validate_metadata
    raise ValueError(
ValueError: 'geo' metadata in Parquet/Feather file is missing required key 'crs' for column 'geometry'
```

`crs` field is optional in recent versions of GeoParquet standard, and setting it as `null` has a different meaning from omitting it. We can omit `crs` by default. This requires us to upgrade geopandas to 0.13.2, and drop support for Python 3.7 since geopandas dropped support for Python 3.7 since 0.11.

The value for `geoparquet.crs` or `geoparquet.crs.<column_name>` option user specified can be one of the following values:
* `""` (empty string): omit the `crs` metadata
* `"null"`: explicitly setting `crs` as `null`
* `"{...PROJJSON...}"`: explicitly setting `crs` to specified PROJJSON object

Reply:
This sounds good to me but GeoPandas > 0.11 requires shapely 2.0, and our other Python functions will not work in this case.

So maybe we still set the crs `null` by default and change the default behavior in the next major release 1.6.0, in which we will add Shapely 2.0 support?

Other parts look good to me:

```
The value for geoparquet.crs or geoparquet.crs.<column_name> option user specified can be one of the following values:

"" (empty string): omit the crs metadata
"null": explicitly setting crs as null
"{...PROJJSON...}": explicitly setting crs to specified PROJJSON object

```

Reply:
GeoPandas 0.13.2 works with shapely >=1.7.1, so it does not require shapely 2.0. I've confirmed that it works fine with shapely 1.8.5 and could pass all the tests after applying https://github.com/apache/sedona/pull/1163.

If we decide to upgrade GeoPandas and drop Python 3.7 in the next release, we can keep setting `crs` as `null` by default for now, and allow user to explicitly omit `crs` by setting `geoparquet.crs` to `""`.

Reply:
Yes, let's postpone the GeoPandas upgrade to 1.6.0. Let's keep setting crs as null by default for now, and allow user to explicitly omit crs by setting geoparquet.crs to ""